ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateTransition.java

```diff
@@ -57,16 +57,15 @@ public void initiate(BeaconStateWithCache state, BeaconBlock block)
             .mod(UnsignedLong.valueOf(SLOTS_PER_EPOCH))
             .equals(UnsignedLong.ZERO)) {
       epochProcessor(state, block);
+      // Client specific optimization
+      state.invalidateCache();
     }
 
     slotProcessor(state);
 
-    if (block != null) {
+    if (Objects.nonNull(block)) {
```

Comment:
Can you expound on the necessity of `Objects.nonNull()`

Reply:
I guess there is no real difference.  I almost did `block = null` and it annoyed me so i used the Objects wrapper.